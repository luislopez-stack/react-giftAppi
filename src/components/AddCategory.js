import React, {useState} from 'react'
import PropTypes from 'prop-types'

function AddCategory({ setCategories }) {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }


    const handleSubmit = (event) => {
        event.preventDefault();
        setCategories(cats => [inputValue, ...cats]);
    }


  return (
      <>
        <form onSubmit={handleSubmit}>

            <h4>{inputValue}</h4>
            <input
                type="text"
                vaue={inputValue}
                onChange={handleInputChange}/>

        </form>
      </>
  )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory
