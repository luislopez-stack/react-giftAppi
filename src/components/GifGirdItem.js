import React from 'react'
import PropTypes from 'prop-types'

export function GifGirdItem({id,title,url}) {
  return (
    <div className='card animate__fadeIn animate__animated'>
        <img src={url} alt={title}></img>
        <p>{title}</p>
    </div>
  )
}

GifGirdItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default GifGirdItem
