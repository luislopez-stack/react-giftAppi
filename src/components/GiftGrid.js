import React, {useEffect, useState} from 'react'
import PropTypes from 'prop-types'
import GifGirdItem from './GifGirdItem';
import { useEffectGift } from '../hooks/useEffectGift';

const GiftGrid = ({category}) => {


  const {loading, data} = useEffectGift(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && <p>Loading</p>}

      <div className='card-grid'>
          
            {
              data.map( img => (
                <GifGirdItem 
                key={img.id}
                {...img}
                />
                ))
              }
          
      </div>
    </>
  )
}

GiftGrid.propTypes = {}

export default GiftGrid 