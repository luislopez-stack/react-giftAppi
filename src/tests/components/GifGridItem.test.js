import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import {GifGirdItem} from '../../components/GifGirdItem';


describe('Pruebas en GiftGridItem', () => {
    
    const title = 'Titulo';
    const url = 'http:localhost';
    const wrapper = shallow(<GifGirdItem title={title} url={url}/>)

    test('should mostrar componente correctamente', () => { 
        
        expect(wrapper).toMatchSnapshot();
     })


     test('should tener parrafo con titlo', () => { 
         const p = wrapper.find('p');
         expect(p.text().trim()).toBe(title);
      })

    
      test('should tener imagen', () => { 
        const img = wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
     })


     test('should tener la clase el div', () => { 
         const div = wrapper.find('div');
         //console.log(div.props().className);
         expect(div.props().className.includes('animate__fadeIn')).toBe(true)
      })
})
