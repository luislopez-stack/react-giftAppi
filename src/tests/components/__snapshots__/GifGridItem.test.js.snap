// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftGridItem should mostrar componente correctamente 1`] = `
<div
  className="card animate__fadeIn animate__animated"
>
  <img
    alt="Titulo"
    src="http:localhost"
  />
  <p>
    Titulo
  </p>
</div>
`;
